src/mrmimeparser.c

```diff
@@ -1024,8 +1027,55 @@ static int mrmimeparser_add_single_part_if_known(mrmimeparser_t* ths, struct mai
 					part = mrmimepart_new();
 					part->m_type = MR_MSG_TEXT;
 					part->m_int_mimetype = mime_type;
-					part->m_msg = simplified_txt;
-					part->m_msg_raw = strndup(decoded_data, decoded_data_bytes);
+					
+                    /* cs:
```

Comment:
m_msg needs to be modified because this is displayed to the user, however, m_msg may contain modified data that cannot be uudecoded properly. all data are in m_msg_raw.

thinking it over, I think the best way is to do the uuencode some lines above before mrsimplify_simplify() is called; on decoded_data and decoded_data_bytes then (CAVE: we should check if we can modify the buffers directly).

Reply:
Yes, that's true:
 As the new functions are designed to cut out the uuencoded part from the message they should work on *decoded_data*. This means before the message is doubled to *m_msg* and *m_msg_raw*.
 All other functions can stay then as they are if the encoded part is cutted out early enough.

I would prefer to have a link in remaining message text to the attachment if possible. At least a hint with the filename at the cut position. This guarantees the relation to the original text position.

By the way: what means CAVE ?

Reply:
In general it should be possible to modify the buffer directly because the resulting text in each case is shorter or in maximum unchanged if no uuencoded part is found!